libraries/profile.rb

```diff
@@ -141,6 +141,22 @@ class ComplianceProfile < Chef::Resource
         end
       end
 
+      action :upload do
+        converge_by 'upload compliance profile' do
+          raise 'unable to determine path to compressed profile archive' if path.nil?
+          o, p = normalize_owner_profile
+          node.run_state['compliance'] ||= {}
+
+          if node.run_state['compliance']['access_token']
+            reqpath ="owners/#{o}/compliance/#{p}/tar"
+            url = construct_url(server, reqpath)
+            Chef::Log.info "Upload compliance profile #{o}/#{p} to: #{url}"
+            res = Compliance::HTTP.post_file(url.to_s, node.run_state['compliance']['access_token'], path, insecure)
```

Comment:
@chris-rock  I agree, using `Compliance::API.upload()` is definitely the way to go.  I'll make that change easily.

The error checking would be a nice to have.  I'm thinking that refactoring the `inspec-compliance` bundle to pull the error checking out of  `ComplianceCli` class would be the longer term solution. The tricky part being to make it non `cli` specific and more API friendly.

What if as a first stab, we duplicate some of the rudimentary error checking from above in the `audit` cookbook resource? Like:

1. path exist
2. profile.check
3. profile exists? (overwrite should probably be the default for a build-cookbook scenario?)